From 42bf6af7461db84ec123573298818822eed5bb88 Mon Sep 17 00:00:00 2001
From: George Chang <georgekgchang@google.com>
Date: Wed, 5 Jun 2019 14:27:50 +0800
Subject: [PATCH 1/3] Prevent OOB write in phFriNfc_ExtnsTransceive

Bug: 132083376
Test: Manual test/ Read/Write Mifare Tag
Merged-In: Ic6e5053b369ead645293c94854431b1c17ed6431
Change-Id: Ic6e5053b369ead645293c94854431b1c17ed6431
(cherry picked from commit 1d3d3748b8c6bc9850153ac25cffa434e46ea25c)
---
 nci/jni/extns/pn54x/src/mifare/phNxpExtns_MifareStd.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/nci/jni/extns/pn54x/src/mifare/phNxpExtns_MifareStd.c b/nci/jni/extns/pn54x/src/mifare/phNxpExtns_MifareStd.c
index 6f58e08c..caacc458 100755
--- a/nci/jni/extns/pn54x/src/mifare/phNxpExtns_MifareStd.c
+++ b/nci/jni/extns/pn54x/src/mifare/phNxpExtns_MifareStd.c
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 
+#include <log/log.h>
 #include <nfc_api.h>
 #include <rw_api.h>
 #include <phNfcCompId.h>
@@ -2051,6 +2052,12 @@ NFCSTATUS phFriNfc_ExtnsTransceive(phNfc_sTransceiveInfo_t *pTransceiveInfo,
     uint32_t length = SendLength;
     uint8_t restore_payload[]={0x00, 0x00, 0x00, 0x00,};
 
+    if( SendLength == 0 )
+    {
+        android_errorWriteLog(0x534e4554, "132083376");
+        return status;
+    }
+
     buff = (uint8_t *)malloc((uint32_t)MAX_BUFF_SIZE);
     if( NULL == buff )
     {
-- 
2.17.1

